src/plugin.js

```diff
@@ -261,6 +261,11 @@ const contribAdsPlugin = function(options) {
           player.ads.adTimeoutTimeout = window.setTimeout(function() {
             player.trigger('adtimeout');
           }, settings.prerollTimeout);
+          // We will try to pause content here. This is because in an autoadvance playlist
+          // on Android devices with HLS renditions, the duration is incorrectly reported
+          // as infinity, due to which the content is not paused incorrectly
+          // So before we signal to the ad plugin , we want to pause content
+          cancelContentPlay(player);
```

Comment:
There's a pretty obvious follow-up question to that: why do we choose not to pause when duration is Infinity? The answer is that for many live video setups with client-side ads, we want the video to continue playing in the background so that we can be notified when the ad break is completing. This change would break those setups.